### PR TITLE
Properly handle `@ApplicationPath` and `Application` in RESTEasy Classic

### LINKED ISSUE #42963: RESTEasy Classic - annotation on abstract jakarta.ws.rs.core.Application class is not ignored
### Describe the bug

RESTEasy Classic - annotation on abstract jakarta.ws.rs.core.Application class is not ignored

https://github.com/quarkusio/quarkus/pull/41465 should ensure abstract jakarta.ws.rs.core.Application classes are ignored

But it's not the case when I put `@ApplicationPath` on it. Path gets unfortunately reflected.
E.g. endpoint is available on http://localhost:8080/rest/hello instead of http://localhost:8080/hello.

### Expected behavior

Annotation on abstract jakarta.ws.rs.core.Application class is ignored

### Actual behavior

annotation on abstract jakarta.ws.rs.core.Application class is NOT ignored

### How to Reproduce?

 - Get app https://code.quarkus.io/?e=resteasy&extension-search=resteasy%20classic
 - add AbstractApplication into the app
 - run `mvn clear verify`
 - alternatively run `mvn quarkus:dev` and access http://localhost:8080/rest/hello and http://localhost:8080/hello

```java
package org.acme;

import jakarta.ws.rs.ApplicationPath;
import jakarta.ws.rs.core.Application;

@ApplicationPath("/rest")
public abstract class AbstractApplication extends Application {
}
```

### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

Java 21

### Quarkus version or git rev

Quarkus 3.14.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
`@ApplicationPath` is only taken into account for the single `Application` class that is selected (if any).

- Fixes: #42963

### CODE REVIEW

File: extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java [hunk 1]
```diff
@@ -1098,49 +1096,37 @@ private static JaxrsProvidersToRegisterBuildItem getFilteredJaxrsProvidersToRegi
...
-            if (selectedAppClass.annotationsMap().containsKey(ResteasyDotNames.CDI_INJECT)) {
-                throw new RuntimeException(
-                        "Usage of '@Inject' is not allowed in 'jakarta.ws.rs.core.Application' classes. Offending class is '"
-                                + selectedAppClass.name() + "'");
-            }
+        if (jakartaRestApplicationClass.annotationsMap().containsKey(ResteasyDotNames.CDI_INJECT)) {
+            throw new RuntimeException(
+                    "Usage of '@Inject' is not allowed in 'jakarta.ws.rs.core.Application' classes. Offending class is '"
+                            + jakartaRestApplicationClass.name() + "'");
+        }
```
Comment (jamezp) [inline extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:1109]: Not the point of this PR, but per the spec an application type should be a CDI bean with a `@ApplicationScoped` scope. There could be other technical reasons this is done in Quarkus, but just an FYI.

**Review by jamezp — APPROVED** [review]

File: extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:1109]: Thanks for bringing that up. I don't remember why we did this, but it's certainly something to follow up on later

Comment (geoand) [reply extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:1110]: This was introduced in https://github.com/quarkusio/quarkus/issues/22277. We should definitely revisit that

File: extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java [hunk 2]
```diff
@@ -222,44 +222,44 @@ public void build(
...
+                    jaxrsProvidersToRegisterBuildItem, allowedClasses, excludedClasses);
         }
 
         boolean filterClasses = !allowedClasses.isEmpty() || !excludedClasses.isEmpty();
 
-        // currently we only examine the first class that is annotated with @ApplicationPath so best
-        // fail if the user code has multiple such annotations instead of surprising the user
-        // at runtime
-        if (applicationPaths.size() > 1) {
-            throw createMultipleApplicationsException(applicationPaths);
```
Comment (rsvoboda) [inline extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:260]: `createMultipleApplicationsException` is not called anymore, should be also dropped

**Review by rsvoboda — CHANGES_REQUESTED** [review]
Tested it with the reproducer from https://github.com/quarkusio/quarkus/issues/42963 and works as expected with this fix.

Small change needed to this PR.

File: extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java (hunk 2 above)
Comment (geoand) [reply extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:260]: Good catch! It's now fixed

**Review by rsvoboda — APPROVED** [review]

File: extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java (hunk 1 above)
Comment (rsvoboda) [reply extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:1095]: > We should definitely revisit that

Would be good to have GH issue for it ;)

Comment (geoand) [reply extensions/resteasy-classic/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:1095]: https://github.com/quarkusio/quarkus/issues/44218

**Pull request merged**